pkg/addon/create.go

```diff
@@ -0,0 +1,264 @@
+func CheckAddonName(addonName string) error {
+	if len(addonName) == 0 {
+		return fmt.Errorf("addon name should not be empty")
+	}
+
+	// Make sure addonName only contains lowercase letters, dashes, and numbers, e.g. some-addon
+	re := regexp.MustCompile(`^[a-z\d]+(-[a-z\d]+)*$`)
+	if !re.MatchString(addonName) {
+		return fmt.Errorf("addon name should only cocntain lowercase letters, dashes, and numbers, e.g. some-addon")
+	}
+
+	return nil
+}
+
+// writeHelmComponentTemplate writes a cue, with a helm component inside, intended as addon resource
+func writeHelmComponentTemplate(tmpl HelmComponentTemplate, filePath string) error {
+	r := cue.Runtime{}
+	v, err := gocodec.New(&r, nil).Decode(tmpl)
+	if err != nil {
+		return err
+	}
+
+	// Use `output` value
+	v = v.Lookup("output")
+	// Format output
+	bs, err := format.Node(v.Syntax())
+	if err != nil {
+		return err
+	}
+
+	// Append "output: " to the beginning of the string, like "output: {}"
+	bs = append([]byte("output: "), bs...)
+	err = os.WriteFile(filePath, bs, 0644)
+	if err != nil {
+		return fmt.Errorf("cannot write %s: %w", filePath, err)
+	}
+
+	return nil
+}
+
+// createAddonFiles creates the file structure for an addon,
+// including template.yaml, readme.md, metadata.yaml, and <addon-nam>.cue.
+func createAddonFiles(addonPath, addonName, helmRepoURL, chartName, chartVersion string) error {
+
+	// TODO(charlie0129): fill out some basic info in the template for the user (read them from the chart), like chart logo, description, and readme, if the chart exists
+
+	// Write README.md
+	readmeFilePath := path.Join(addonPath, ReadmeFileName)
+	err := os.WriteFile(readmeFilePath,
+		[]byte(fmt.Sprintf("# %s\nInsert the README of your addon here.\n\nAlso check how to build your own addon: https://kubevela.net/docs/platform-engineers/addon/intro", addonName)),
+		0644)
+	if err != nil {
+		return fmt.Errorf("cannot write %s: %w", readmeFilePath, err)
+	}
+
+	// Write template.yaml with an empty Application
+	templateFilePath := path.Join(addonPath, TemplateFileName)
+	applicationTemplate := v1beta1.Application{
+		TypeMeta: v1.TypeMeta{
+			APIVersion: v1beta1.SchemeGroupVersion.String(),
+			Kind:       "Application",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      addonName,
+			Namespace: types.DefaultKubeVelaNS,
+		},
+	}
+
+	applicationTemplateBytes, err := yaml.Marshal(applicationTemplate)
+	if err != nil {
+		return err
+	}
+	err = os.WriteFile(templateFilePath,
+		applicationTemplateBytes,
+		0644)
+	if err != nil {
+		return fmt.Errorf("cannot write %s: %w", templateFilePath, err)
+	}
+
+	// Write metadata.yaml with `fluxcd` as a dependency because we are using helm.
+	// However, this may change in the future, possibly with `argocd`.
+	metadataFilePath := path.Join(addonPath, MetadataFileName)
+	metadataTemplate := Meta{
+		Name:        addonName,
+		Version:     chartVersion,
+		Description: "Insert the description of your addon here.",
```

Comment:
`Description: "An addon for KubeVela" ` ?

Reply:
changes made